tests/kernels/test_lightning_attn.py

```diff
@@ -0,0 +1,328 @@
+        v_b = v[b, :, 0]  # [H, D]
+
+        # Process each attention head
+        for h in range(H):
+            # Get current query, key and value
+            q_bh = q_b[h]
+            k_bh = k_b[h]
+            v_bh = v_b[h]
+
+            # Get cache
+            kv_cache_old = kv_caches[b, h]
+
+            # Calculate new key-value outer product
+            kv_outer = torch.outer(k_bh, v_bh)
+
+            # Apply decay and update cache
+            kv_new = kv_outer + decay[h, 0, 0] * kv_cache_old
+
+            # Calculate output
+            out_h = torch.matmul(q_bh, kv_new)
+
+            # Update output and cache
+            output[b, h * D:(h + 1) * D] = out_h
+            kv_caches[b, h] = kv_new
+
+    return output
+
+
+@pytest.mark.parametrize("batch_size", BATCH_SIZES)
+@pytest.mark.parametrize("num_heads", NUM_HEADS)
+@pytest.mark.parametrize("head_size", HEAD_SIZES)
+@pytest.mark.parametrize("dtype", DTYPES)
+@torch.inference_mode()
+def test_linear_decode_forward_triton(
+    batch_size: int,
+    num_heads: int,
+    head_size: int,
+    dtype: torch.dtype,
+):
+    torch.set_default_device("cuda")
+    torch.manual_seed(42)
+    torch.cuda.manual_seed_all(42)
+    current_platform.seed_everything(42)
+
+    base = 0.01
+    q = torch.zeros(batch_size, num_heads, 1, head_size, dtype=dtype)
+    k = torch.zeros(batch_size, num_heads, 1, head_size, dtype=dtype)
+    v = torch.zeros(batch_size, num_heads, 1, head_size, dtype=dtype)
+
+    for i in range(batch_size):
+        for j in range(num_heads):
+            for d in range(head_size):
+                q[i, j, 0, d] = base * (i + j + d + 1)
+                k[i, j, 0, d] = base * (i + j + d + 2)
+                v[i, j, 0, d] = base * (i + j + d + 3)
+
+    kv_caches = torch.zeros(batch_size,
+                            num_heads,
+                            head_size,
+                            head_size,
+                            dtype=dtype,
+                            device="cuda")
+
+    for i in range(batch_size):
+        for j in range(num_heads):
+            for k_idx in range(head_size):
+                for v_idx in range(head_size):
+                    kv_caches[i, j, k_idx,
+                              v_idx] = base * (i + j + k_idx + v_idx + 4)
+
+    kv_caches_copy = kv_caches.clone()
+
+    slope_rate = torch.zeros(num_heads, device="cuda")
+    for h in range(num_heads):
+        slope_rate[h] = 0.1 * (h + 1)
+
+    slot_idx = torch.arange(batch_size, device="cuda")
+
+    triton_output = linear_decode_forward_triton(q, k, v, kv_caches,
+                                                 slope_rate, slot_idx)
+
+    reference_output = reference_linear_decode(q, k, v, kv_caches_copy,
+                                               slope_rate, slot_idx)
+    torch.testing.assert_close(triton_output,
+                               reference_output,
+                               rtol=1e-1,
+                               atol=1e-1)
```

Comment:
1e-1 seems pretty high for `rtol` and `atol` here, so wondering if that's related to the above initialization